cpp/nacjd.cc

```diff
@@ -0,0 +1,371 @@
+            const std::string id(alternateIdentifier_node->getStringNode("identifier")->getValue());
+            if (complete) {
+                MARC::Record new_record(MARC::Record::TypeOfRecord::LANGUAGE_MATERIAL, MARC::Record::BibliographicLevel::UNDEFINED, "[ICPSR]" + id);
+                new_record.insertField("245", { { 'a', title_node->getValue() } }, /* indicator 1 = */'0', /* indicator 2 = */'0');
+                new_record.insertField("520", { { 'a', description } });
+                new_record.insertField("540", { { 'a', license } });
+                new_record.insertField("264", { { 'c', initial_release_date } });
+                new_record.insertField("856", { { 'u', "https://www.icpsr.umich.edu/web/NACJD/studies/" + id } });
+                for (auto creator : creators)
+                    new_record.insertField(creator.second, { { 'a', creator.first } });
+                for (auto keyword : keywords) {
+                    const std::string normalized_keyword(TextUtil::CollapseAndTrimWhitespace(keyword));
+                    new_record.insertField(MARC::GetIndexField(normalized_keyword));
+                }
+                title_writer->write(new_record);
+                break;
+            }
+        }
+    }
+    LOG_INFO("Processed: " + std::to_string(no_total) + "entries. " + std::to_string(no_initial_date) + " w/o initial date, " 
+                    + std::to_string(no_title) + " w/o title, " + std::to_string(no_creators) + " w/o creator and " + std::to_string(no_license) + " w/o license.");
+}
+
+
+
+} // unnamed namespace
+
+
+int main(int argc, char *argv[]) {
+    ::progname = argv[0];
+
+    if (argc != 3)
+        Usage();
+
+    auto marc_reader(MARC::Reader::Factory(argv[1]));
+    auto marc_writer(MARC::Writer::Factory(argv[2]));
+
+    try {
+        // parse marc_file (later phase_x) and store ids in set
+        std::set<std::string> parsed_marc_ids;
+        LOG_INFO("Extracting existing ICPSR ids from marc input...");
+        extractExistingIDsFromMarc(marc_reader.get(), parsed_marc_ids);
+        LOG_INFO("Found: " + std::to_string(parsed_marc_ids.size()) + " records with ICPSR ids.");
+
+        // Download possible new publications and store in json file
+        LOG_INFO("Extracting ICPSR ids from website...");
+        unsigned number_of_new_ids;
+        extractIDsFromWebsite(parsed_marc_ids, &number_of_new_ids);
+        LOG_INFO(std::to_string(number_of_new_ids) +  " new ids collected from website.");
+
+        // parse json file and store relevant information in variables
+        // write marc_records via marc_writer to marc_file
+        LOG_INFO("Parsing intermediate json file and save to marc output...");
+        parseJsonAndWriteMarc(marc_writer.get());
+        LOG_INFO("Finished.");
+
+        return EXIT_SUCCESS;
+    } catch (const std::exception &x) {
+
```

Comment:
Missing error message and "return EXIT_FAILURE;" but, you won't need them w/ "Main".